### PR TITLE
Card editing mode

### PR DESCRIPTION
**Pull request by Jaskon**

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -36,13 +36,28 @@ function App() {
...
+            : newOne
+        )
+      )}
+      onEditSave={({ caption, content }) => setCardsState(
+        cardsState.map(newOne =>
+          newOne.id === one.id
+            ? {...newOne, caption, content, editing: false}
+            : newOne
+        )
+      )}
```
Comment (kkaroukinIBA) [inline src/App/App.js]: I think it is redundant code. Good practice will be to pass only one method of manipulation with card.

File: src/Card/Card.js [hunk 2]
```diff
@@ -1,14 +1,69 @@
...
+    }
+  }
+
+  const onSaveClick = e => {
+    e.stopPropagation();
+    if (onEditSave !== undefined) {
+      // onEditSave(state);
+      onEditSave(state);
+    }
+  }
```
Comment (kkaroukinIBA) [inline src/Card/Card.js]: Can you please refactor it according previos comment?

File: src/Card/Card.js [hunk 3]
```diff
@@ -1,14 +1,69 @@
...
+    : <div className={'Card__caption'}>{caption}</div>;
+
+  const contentRendered = editMode
+    ? <textarea
+        className={'Card__input-content'}
+        onChange={e => setState({...state, content: e.target.value})}
+        defaultValue={state.content}
+        />
+    : <div className={'Card__content'}>{content}</div>;
+
```
Comment (kkaroukinIBA) [inline src/Card/Card.js]: You are checking the same three times. Can you please refactor your component to call this only one time?

**Review by kkaroukinIBA — CHANGES_REQUESTED** [review]

File: src/App/App.js [hunk 4]
```diff
@@ -30,19 +30,27 @@ const beValues = [
...
+      editing={card.editing}
       onSelect={() => setCardsState(
         cardsState.map(newOne =>
-          newOne.id === one.id
+          newOne.id === card.id
             ? {...newOne, selected: !newOne.selected}
             : newOne
         )
       )}
+      onEdit={newOne => setCardsState(
```
Comment (kkaroukinIBA) [inline src/App/App.js]: Can you please tell me the difference between workflow of "onSelect" and "onEdit" methods you're passing to Card?
Can you please pass only one method for changing card?

File: src/Card/Card.js [hunk 5]
```diff
@@ -1,12 +1,79 @@
...
 import cl from 'classnames';
 import './Card.scss';
+import { BiEditAlt, BiXCircle, BiSave } from 'react-icons/bi';
 
-function Card({ className, caption, content, onSelect, selected = false }) {
+function Card({ className, caption, content, onSelect, onEdit,
+                editing = false, selected = false }) {
+
+  // New values state
+  const [state, setState] = useState({ caption, content });
```
Comment (kkaroukinIBA) [inline src/Card/Card.js:10]: Can you please rename state value and its setting method?
We don't know sense of "state" name.

**Review by kkaroukinIBA — CHANGES_REQUESTED** [review]

File: src/Card/Card.js (hunk 5 above)
Comment (Jaskon) [reply src/Card/Card.js:10]: Will 'cardState' and 'setCardState' be suitable here?

**Review by kkaroukinIBA — APPROVED** [review]

**Pull request merged**